### PR TITLE
fix: exclude erroneous payments

### LINKED ISSUE #466: Exclude errorneous payments from APIs
When a payment fails (due to e.g. a timeout) a new invoice is created and it's state is set to `Error`. 

API endpoints should not return these kind of transactions as they don't contain any information, anyway:

image

The filtering could take place over here: 

https://github.com/getAlby/lndhub.go/blob/feat/add-transaction-entries-index/lib/service/user.go#L196

### PR DESCRIPTION
**Pull request by reneaaron**
Fixes #466

### CODE REVIEW

File: integration_tests/internal_payment_test.go [hunk 1]
```diff
@@ -399,10 +399,9 @@ func (suite *PaymentTestSuite) TestInternalPaymentFail() {
 		fmt.Printf("Error when getting invoices %v\n", err.Error())
 	}
 
-	// check if first one is settled, but second one error (they are ordered desc by id)
-	assert.Equal(suite.T(), 2, len(invoices))
-	assert.Equal(suite.T(), common.InvoiceStateError, invoices[0].State)
-	assert.Equal(suite.T(), common.InvoiceStateSettled, invoices[1].State)
+	// check if first one is settled, second one is erroneous and hence isn't listed
+	assert.Equal(suite.T(), 1, len(invoices))
+	assert.Equal(suite.T(), common.InvoiceStateSettled, invoices[0].State)
```
Comment (bumi) [inline integration_tests/internal_payment_test.go:404]: can we query for the second one differently and check that it has the state error?

**Review by bumi — COMMENTED** [review]
can we load the other invoice differently and still check that it is there and has the error state?

File: integration_tests/internal_payment_test.go (hunk 1 above)
Comment (im-adithya) [reply integration_tests/internal_payment_test.go]: Fixed

**Review by bumi — APPROVED** [review]

**Comment by reneaaron** [comment]
Follow-up: #469

**Pull request merged**